pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -94,19 +91,26 @@ func (s *Reconciler) Create(ctx context.Context) error {
 		return errors.Wrapf(err, "failed to create vm %s ", s.scope.Machine.Name)
 	}
 
-	scriptData, err := config.GetVMStartupScript(s.scope, bootstrapToken)
-	if err != nil {
-		return errors.Wrapf(err, "failed to get vm startup script")
-	}
+	if s.scope.MachineConfig.UserDataSecret == nil {
+		bootstrapToken, err := s.checkControlPlaneMachines()
+		if err != nil {
+			return errors.Wrap(err, "failed to check control plane machines in cluster")
+		}
 
-	vmExtSpec := &virtualmachineextensions.Spec{
-		Name:       "startupScript",
-		VMName:     s.scope.Machine.Name,
-		ScriptData: base64.StdEncoding.EncodeToString([]byte(scriptData)),
-	}
-	err = s.virtualMachinesExtSvc.CreateOrUpdate(ctx, vmExtSpec)
-	if err != nil {
-		return errors.Wrap(err, "failed to create vm extension")
+		scriptData, err := config.GetVMStartupScript(s.scope, bootstrapToken)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get vm startup script")
+		}
+
+		vmExtSpec := &virtualmachineextensions.Spec{
+			Name:       "startupScript",
+			VMName:     s.scope.Machine.Name,
+			ScriptData: base64.StdEncoding.EncodeToString([]byte(scriptData)),
+		}
+		err = s.virtualMachinesExtSvc.CreateOrUpdate(ctx, vmExtSpec)
```

Comment:
Simplify to 

```go
if err := s.virtualMachinesExtSvc.CreateOrUpdate(ctx, vmExtSpec); err != nil {
...
```

Reply:
unfortunately linter complains of creating a overshadowing variable for if err := ; err != nil pattern hence i went this route :( I dont want to make mass changes since we want to merge upstream